test/e2e/autonomy/autonomy_test.go

```diff
@@ -0,0 +1,330 @@
+		ginkgo.It("kube-proxy edge-autonomy test", ginkgo.Label("edge-autonomy"), func() {
+			// obtain kube-proxy containerID with crictl
+			gomega.Eventually(func() bool {
+				cmd := `docker exec -t openyurt-e2e-test-worker /bin/bash -c "crictl ps | grep kube-proxy | awk '{print \$1}'"`
+				opBytes, err := exec.Command("/bin/bash", "-c", cmd).CombinedOutput()
+				if err != nil {
+					return false
+				}
+				kubeProxyContainerID = strings.Trim(string(opBytes), "\r\n")
+				return true
+			}).Should(gomega.BeTrue(), "fail to get kube-proxy container ID")
+			// restart kube-proxy
+			gomega.Eventually(func() bool {
+				_, err := exec.Command("/bin/bash", "-c", "docker exec -t openyurt-e2e-test-worker /bin/bash -c 'crictl stop "+kubeProxyContainerID+"'").CombinedOutput()
+				if err != nil {
+					return false
+				}
+				return true
+			}).Should(gomega.BeTrue(), "fail to stop kube-proxy")
+
+			gomega.Eventually(func() bool {
+				_, err := exec.Command("/bin/bash", "-c", "docker exec -t openyurt-e2e-test-worker /bin/bash -c 'iptables -F'").CombinedOutput()
+				if err != nil {
+					return false
+				}
+				return true
+			}).Should(gomega.BeTrue(), "fail to remove iptables on node "+YurtEdgeNode2Name)
+
+			gomega.Eventually(func() string {
+				opBytes, err := exec.Command("/bin/bash", "-c", "docker exec -t openyurt-e2e-test-worker /bin/bash -c 'curl "+NginxServiceIP+"'").CombinedOutput()
+				if err != nil {
+					return ""
+				}
+				return string(opBytes)
+			}).WithTimeout(10*time.Second).WithPolling(1*time.Second).Should(gomega.ContainSubstring("nginx"), "fail to read curl response from service: "+NginxServiceName)
+		})
+	})
+
+	var _ = ginkgo.Describe("coredns"+YurtE2ENamespaceName, func() {
+		ginkgo.It("coredns edge-autonomy test", ginkgo.Label("edge-autonomy"), func() {
+			ginkgo.Skip("current coredns does not support edge-autonomy, coredns-edge-autonomy tests will be skipped.")
+			// obtain coredns containerID with crictl
+			gomega.Eventually(func() bool {
+				cmd := ` /bin/bash -c "crictl ps | grep coredns | awk '{print \$1}'"`
+				opBytes, err := exec.Command("/bin/bash", "-c", "docker exec -t "+coreDnsNodeName+cmd).CombinedOutput()
+				if err != nil {
+					return false
+				}
+				coreDnsContainerID = strings.Trim(string(opBytes), "\r\n")
+				return true
+			}).Should(gomega.BeTrue(), "fail to get coredns container ID")
+			// restart kube-proxy
+			gomega.Eventually(func() bool {
+				cmd := "docker exec -t " + coreDnsNodeName + " /bin/bash -c 'crictl stop " + coreDnsContainerID + "'"
+				_, err := exec.Command("/bin/bash", "-c", cmd).CombinedOutput()
+				if err != nil {
+					return false
+				}
+				return true
+			}).Should(gomega.BeTrue(), "fail to stop coredns")
+			gomega.Eventually(func() string {
+				cmd := "docker exec -t openyurt-e2e-test-worker /bin/bash -c 'dig " + "@" + CoreDNSServiceIP + " " + NginxServiceName + "." + YurtDefaultNamespaceName + ".svc.cluster.local'"
```

Comment:
Could we use CoreDNSServiceIP ? We only kill one CoreDNS pod, if using `kube-dns`, will this request be loadbalanced to another healthy CoreDNS Pod?

BTW, please use `fmt.Sprintf` instead of `NginxServiceName + "." + YurtDefaultNamespaceName + ".svc.cluster.local'"`

Reply:
the other CoreDNS pod is on openyurt-e2e-test-control-plane, which is disconnect from kind-net-bridge in BeforeSuite, so the request should not be able to be loadbalanced to it, hence it's okay to use CoreDNSServiceIP.
will change to fmt.Sprintf

Reply:
Will it definitely run on control-plane node, and never be scheduled to edge node ?

Reply:
We'll use kubectl patch in run-e2e-tests.sh, before we run the edge-autonomy tests, to make sure there is one and only one coreDNS on edge. We'll be able to use coreDNSServiceIP in core dns edge-autonomy-tests, you can see the change in new push.